modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_0_0/PollsUpgradeProcess.java

```diff
@@ -0,0 +1,1074 @@
+			modifiedDate, structureId, name, description, expirationDate,
+			lastPublishDate);
+
+		_upgradeResourcePermission(questionId);
+
+		_addDDMFormInstanceVersion(
+			groupId, companyId, userId, userName, createDate, questionId,
+			structureVersionId, name, description, lastPublishDate);
+
+		JSONObject dataJSONObject = getDataJSONObject(ddmFormField);
+
+		_upgradePollsVotes(
+			dataJSONObject, ddmFormFieldOptionsValues, questionId, structureId,
+			structureVersionId, ddmFormField.getName(),
+			String.valueOf(ddmFormField.getProperty("instanceId")));
+
+		_addDDMFormInstanceReport(
+			groupId, companyId, createDate, lastVoteDate, questionId,
+			dataJSONObject);
+	}
+
+	private void _upgradePollsQuestions() throws Exception {
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"select * from PollsQuestion");
+			ResultSet resultSet = preparedStatement.executeQuery()) {
+
+			while (resultSet.next()) {
+				long questionId = resultSet.getLong("questionId");
+
+				String title = resultSet.getString("title");
+
+				_availableLocales = SetUtil.fromArray(
+					LocaleUtil.fromLanguageIds(
+						LocalizationUtil.getAvailableLanguageIds(title)));
+				_defaultLocale = LocaleUtil.fromLanguageId(
+					LocalizationUtil.getDefaultLanguageId(title));
+
+				_upgradePollsQuestion(
+					questionId, resultSet.getLong("groupId"),
+					resultSet.getLong("companyId"), resultSet.getLong("userId"),
+					resultSet.getString("userName"),
+					resultSet.getTimestamp("createDate"),
+					resultSet.getTimestamp("modifiedDate"),
+					_getName(resultSet.getString("title")),
+					resultSet.getString("description"),
+					resultSet.getTimestamp("expirationDate"),
+					resultSet.getTimestamp("lastPublishDate"),
+					resultSet.getTimestamp("lastVoteDate"));
+			}
+		}
+	}
+
+	private void _upgradePollsVote(
+			long groupId, long companyId, long userId, String userName,
+			Timestamp createDate, Timestamp modifiedDate, long formInstanceId,
+			long structureId, long structureVersionId,
+			Timestamp lastPublishDate, String fieldName, String instanceId,
+			String optionValue)
+		throws Exception {
+
+		long storageId = _addDDMStorageLink(
+			companyId, structureId, structureVersionId);
+
+		_addLanguageDDMField(companyId, storageId, structureVersionId);
+
+		_addRadioDDMField(
+			companyId, storageId, structureVersionId, fieldName, instanceId,
+			optionValue);
+
+		long formInstanceRecordId = _addDDMFormInstanceRecord(
+			groupId, companyId, userId, userName, createDate, modifiedDate,
+			formInstanceId, storageId, lastPublishDate);
+
+		_addDDMFormInstanceRecordVersion(
+			groupId, companyId, userId, userName, createDate, formInstanceId,
+			formInstanceRecordId, storageId, lastPublishDate);
+	}
+
+	private void _upgradePollsVotes(
+			JSONObject dataJSONObject,
+			Map<Long, String> ddmFormFieldOptionsValues, long questionId,
+			long structureId, long structureVersionId, String fieldName,
+			String instanceId)
```

Comment:
@natocesarrego I was trying to prioritize the `table columns order`, but now that you mention it, `alphabetical order` makes more sense in this case because `long questionId, long structureId, long structureVersionId` are columns from another table. I'll change it.